docs/reference/migration/migrate_5_0/suggest.asciidoc

```diff
@@ -25,35 +25,22 @@ to suggestion entries for both context and completion suggesters.
 
 ==== Completion suggester is document-oriented
 
-Suggestions are aware of the document they belong to. This enables
-retrieving any field value from the document. This is exposed
-through the query-time `payload` option in `completion` and `context`
-suggesters:
-
-[source,sh]
----------------
-GET /my_index/_search
-{
-  "suggest": {
-    "fooSuggestion": {
-       "text": "f"
-       "completion": {
-         "field": "fooSuggest",
-         "payload": ["field1", "field2"]
-       }
-     }
-  }
-}
----------------
+Suggestions are aware of the document they belong to. Now, associated
+documents (`_source`) are returned as part of `completion` suggestions.
 
 Previously, `context` and `completion` suggesters supported an index-time
 `payloads` option, which was used to store and return metadata with suggestions.
-Now metadata can be stored as a field in the same document as the
-suggestion for enabling retrieval at query-time. The support for
-index-time `payloads` has been removed to avoid bloating the in-memory
-index with suggestion metadata. The time that it takes to retrieve payloads
-depends heavily on the size of the `_source` field.  The smaller the `_source`,
-the faster the retrieval.
+Now metadata can be stored as part of the the same document as the
+suggestion for retrieval at query-time. The support for index-time `payloads`
+has been removed to avoid bloating the in-memory index with suggestion metadata.
+
+In case of completion queries spanning more than one shard, the suggest is executed
+in two phases, where the last phase fetches the relevant documents from shards,
+implying executing completion requests against a single shard is more performant
+due to the document fetch overhead when the suggest spans multiple shards.
+To get best performance for completions, it is recommended to index completions
```

Comment:
I think this should be a "NOTE".

> NOTE: Because the Completion Suggester will return the full `_source`, the size of the `_source` can impact performance while fetching it from disk and passing it around the network. Therefore, the best performance is going to be achieved with smaller `_source` documents. See the example below using `_source` filtering to avoid returning entire documents unnecessarily. (**add example**)